Applications/Utils/ModelPreparation/ConvertVtkDataArrayToVtkDataArray.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ *
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <numeric>
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+
+#include "MeshLib/Mesh.h"
+
+template <typename ORIG, typename NEW_T>
+void castPropertyVectorToPropertyVector(
+    MeshLib::Properties& properties,
+    std::string const& property_vector_name_in,
+    std::string const& property_vector_name_out)
+{
+    auto const* const orig_property = properties.getPropertyVector<ORIG>(
+        property_vector_name_in, MeshLib::MeshItemType::Cell, 1);
+    auto* new_pv = properties.createNewPropertyVector<NEW_T>(
```

Comment:
Please check for nullptr before access. Also for orig_property.

Reply:
'getOrCreateMeshProperty' needs a mesh, but I unpacked the 'Properties' object from the mesh outside of this function and as a consequence I haven't the mesh here available.